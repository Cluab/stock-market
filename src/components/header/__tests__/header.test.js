import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import Header from '../header';
import valueReducer from '../../../redux/companies/companies';

const store = configureStore({ reducer: { stock: valueReducer } });
const T = true;
describe('header component test cases', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header
              showBackButton={T}
            />
          </Router>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test for back icon visibility', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header
            showBackButton={T}
          />
        </Router>

      </Provider>,
    );
    const icon = document.querySelector('.fa-solid');

    expect(icon).toBeVisible();
  });

  test('test for back icon visibility', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header
            showBackButton={false}
          />
        </Router>

      </Provider>,
    );
    const icon = document.querySelector('.fa-solid');

    expect(icon).toBe(null);
  });
});
