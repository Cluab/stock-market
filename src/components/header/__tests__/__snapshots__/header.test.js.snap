// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component test cases should match with snapshot 1`] = `
<header
  className="header"
  data-testid="header"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa-solid fa-gear"
    />
  </button>
  <h1>
    stock market
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="fa-solid fa-chevron-left test"
    />
  </a>
</header>
`;
